### PR TITLE
(ci): #5670 add GHA for generating docs

### LINKED ISSUE #5670: Generate JSON documentation
With the following PR: https://github.com/vegaprotocol/vega/pull/5667 the protobufs definitions are being imported in the core repo.

In order to limit cluttering the repo, we stopped generating json definitions and swagger. However this is still needed by the docs team to generate the website.

## Tasks
- [ ] add a step in the pipeline to generate all json and swagger
- [ ] Create a new repository dedicated to the documentation codegen
- [ ] On merge to develop, push all update json and swagger on the repo

### PR DESCRIPTION
**Pull request by mkjmdski**
Closes https://github.com/vegaprotocol/vega/issues/5670

### CODE REVIEW

File: .github/workflows/release-docs.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        run: cd vega; bash script/gettools.sh
+      -
+        name: Generate docs
+        run: cd vega; make proto_docs; echo "generated docs:"; find protos/generated -type f
+      -
+        name: Make PR to the docs repo
+        env:
+          GH_TOKEN: ${{ secrets.VEGA_CI_BOT_GITHUB_TOKEN }}
+        run: |
+          mkdir -p documentation/specs/$GITHUB_REF_NAME
```
Comment (edd) [inline .github/workflows/release-docs.yml:47]: In https://github.com/vegaprotocol/documentation/pull/216, the folder name comes out as '#issuenumber' not 'v0.54.0'

File: .github/workflows/release-docs.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          cd documentation
+          git config --global user.email "vega-ci-bot@vega.xyz"
+          git config --global user.name "vega-ci-bot"
+          branch_name="$(date +%d-%m-%Y--%H-%M)-spec-update"
+          git checkout -b "$branch_name"
+          commit_msg="Automated specs update from vegaprotocol/vega#$GITHUB_SHA"
+          git add specs
+          git commit -m "$commit_msg"  || exit 0
+          git push -u origin "$branch_name"
+          pr_url="$(gh pr create --title '$commit_msg to $GITHUB_REF_NAME' --body 'This is automated pull request that will be merged after pipeline passes')"
```
Comment (edd) [inline .github/workflows/release-docs.yml:59]: This is not interpreting correctly, https://github.com/vegaprotocol/documentation/pull/216 has the title '$commit_msg to $GITHUB_REF_NAME'

File: .github/workflows/release-docs.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        env:
+          GH_TOKEN: ${{ secrets.VEGA_CI_BOT_GITHUB_TOKEN }}
+        run: |
+          mkdir -p documentation/specs/$GITHUB_REF_NAME
+          cp -r vega/protos/generated/* documentation/specs/$GITHUB_REF_NAME/
+          ls -al documentation/specs/$GITHUB_REF_NAME/
+          cd documentation
+          git config --global user.email "vega-ci-bot@vega.xyz"
+          git config --global user.name "vega-ci-bot"
+          branch_name="$(date +%d-%m-%Y--%H-%M)-spec-update"
```
Comment (edd) [inline .github/workflows/release-docs.yml:53]: Could this change to 'spec-update-[tag]'? i.e. 'spec-update-v0.54.1'

**Review by edd — CHANGES_REQUESTED** [review]

**Comment by edd** [comment]
An additional requirement (which you can refuse if you like, it's expanding the scope): Could this *also* add https://github.com/vegaprotocol/vega/blob/develop/datanode/gateway/graphql/schema.graphql in the pull request?

File: .github/workflows/release-docs.yml (hunk 1 above)
Comment (mkjmdski) [reply .github/workflows/release-docs.yml:47]: GITHUB_REF_NAME is branch name of tag name depending on the event, in this PR branch name is issue number, that's why it confuses. I will just parse `git tag` for that instead

File: .github/workflows/release-docs.yml (hunk 3 above)
Comment (mkjmdski) [reply .github/workflows/release-docs.yml:53]: We can do it but I would opt for a timestamp for a reason that we might do few updates to the same tag as we want to trigger this action on push to develop

File: .github/workflows/release-docs.yml (hunk 2 above)
Comment (mkjmdski) [reply .github/workflows/release-docs.yml:59]: Yeah I will fix the quotes

**Comment by mkjmdski** [comment]
todo: add assignees -> edd, candida-d

File: .github/workflows/release-docs.yml (hunk 3 above)
Comment (edd) [reply .github/workflows/release-docs.yml:53]: @mkjmdski when we chatted about this I opted for on tag instead - in which case would tag be ok here?

**Comment by edd** [comment]
> todo: add assignees -> edd, candida-d

This is what we agreed earlier, but instead I made a new Github team, @vegaprotocol/docs, which means we won't need to change the workflow as the team members who should be notified of the PR change

**Review by edd — DISMISSED** [review]
Thanks @mkjmdski, this looks great - and thanks for helping me find the --assign flag. Turns out I needed to scroll right.

**Review by edd — DISMISSED** [review]

File: script/gettools.sh [hunk 4]
```diff
@@ -2,10 +2,47 @@
...
+if ! command -v protoc; then
+	echo "protoc not detected in the path"
+	echo "Installing protoc to the version ${PROTOC_VERSION}!!"
+	(
+		cd /tmp
+		wget --quiet "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/${file_name}"
+		unzip "${file_name}"
+		mv bin/protoc /usr/local/bin/
+	)
+fi
```
[the remark this answers is not in the document]
Comment (mkjmdski) [reply script/gettools.sh:24]: I decided to re-use this script to don't repeat dependencies install. CI Agent's do not have protoc installed so I wrote this small if to install it there.

[the remark this answers is not in the document]
Comment (jeremyletang) [reply script/gettools.sh:24]: My point is that we should not need protoc?

[the remark this answers is not in the document]
Comment (jeremyletang) [reply script/gettools.sh:24]: Very well, can we see what commands where failing, protoc should be unnecessary in the codebase.

**Review by edd — DISMISSED** [review]

File: script/gettools.sh [hunk 5]
```diff
@@ -1,16 +1,21 @@
...
+	BIN="/usr/local/bin" && \
+	BUF_VERSION="1.7.0" && \
+	curl -sSL \
+		"https://github.com/bufbuild/buf/releases/download/v${BUF_VERSION}/buf-$(uname -s)-$(uname -m)" \
+		-o "${BIN}/buf" && \
+	chmod +x "${BIN}/buf"
 fi
 
+buf --version
+which buf
```
Comment (jeremyletang) [inline script/gettools.sh:17]: Just looking at this, not sure this is necesary as well seeing it's in the list of things we install, see line 19

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**